src/cogs/rules.py

```diff
@@ -59,19 +60,79 @@ async def on_member_join(self, member):
 
         current_time = time.time()
 
-        self._recent_joins = [
-            x for x in self._recent_joins
-            if current_time - x["join_time"] <= self.file["massjoin_window"]
-        ]
+        if not self.massjoin_active:
+            self._recent_joins = [
+                x for x in self._recent_joins
+                if current_time - x["join_time"] <= self.file["massjoin_window"]
+            ]
+
+        self._recent_joins.append({"join_time": current_time, "id": member.id,
+            "assumed_bot": (
+                ( member.default_avatar_url == member.avatar_url if self.file["massjoin_default_pfp"] else True )
+                and ( time.time() - member.created_at.timestamp() < self.file["massjoin_min_acc_age_val"]
+                    if self.file["massjoin_min_acc_age"] else True ) )
+            })
 
-        self._recent_joins.append({"join_time": current_time, "id": member.id})
 
         join_amount = len(self._recent_joins)
 
-        if join_amount >= self.file["massjoin_amount"] and self.next_massjoin_notif <= current_time:
-            first_joined_id = self._recent_joins[0]["id"]
-            await self._notify_staff(member.guild, "Mass member join detected. " + f"First join ID: `{first_joined_id}`")
-            self.next_massjoin_notif = current_time + self.file["massjoin_notif_timeout"]
+        if join_amount >= self.file["massjoin_amount"] and not self.massjoin_active:
+            self.massjoin_active = True
+
+            cross = "❎"
+            checkmark = "✅"
+
+            msg = await self._notify_staff(member.guild,
+                    f"Mass member join detected. React with {checkmark} to take action or with {cross} to ignore")
+
+            await msg.add_reaction(cross)
+            await msg.add_reaction(checkmark)
+
+            def _check(reaction, user, msg):
+                return ( reaction.message.id == msg.id
+                        and any(role.id == self.file["staff_role"] for role in user.roles)
+                        and user.id != self.bot.user.id )
+
+            reaction, user = await self.bot.wait_for('reaction_add',
+                    check=lambda reaction, user: _check(reaction, user, msg),
+                    timeout=self.file["massjoin_notif_timeout"])
+
+            if reaction.emoji == cross:
+                await msg.reply("Not taking any action and resetting the join detection")
+
+            if reaction.emoji == checkmark:
+                reply = await msg.reply("Users assumed to be bots:\n" + ",\n".join(map(lambda x: f"<@{x['id']}>",
+                    filter(lambda x: x["assumed_bot"], self._recent_joins)))
+                    + "\nUsers assumed to not be bots:\n" + ",\n".join(map(lambda x: f"<@{x['id']}>",
+                    filter(lambda x: not x["assumed_bot"], self._recent_joins))))
```

Comment:
This might not fit in a message, use an IO file and send the file

Reply:
This has the disadvantage of not having the users in the form of pings allowing to access them and get the id.
Adding to this, 2000 characters can actually be quite a few pings.
Since that doesn't ensure everything fits into the message, I'm going to add something to auto-split it into multiple messages.